src/test/java/net/sf/jabref/exporter/MSBibExportFormatTestFiles.java

```diff
@@ -0,0 +1,104 @@
+package net.sf.jabref.exporter;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.TransformerFactoryConfigurationError;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+import org.w3c.dom.Document;
+import org.xml.sax.SAXException;
+
+import com.google.common.base.Charsets;
+import net.sf.jabref.BibDatabaseContext;
+import net.sf.jabref.Globals;
+import net.sf.jabref.JabRefPreferences;
+import net.sf.jabref.MetaData;
+import net.sf.jabref.importer.OutputPrinterToNull;
+import net.sf.jabref.importer.fileformat.BibtexImporter;
+import net.sf.jabref.model.database.BibDatabase;
+import net.sf.jabref.model.entry.BibEntry;
+
+@RunWith(Parameterized.class)
+public class MSBibExportFormatTestFiles {
+
+    public static BibDatabaseContext databaseContext;
+    public static Charset charset;
+    public List<BibEntry> entries;
+    public File tempFile;
+    public static MSBibExportFormat msBibExportFormat;
+    public BibtexImporter testImporter = new BibtexImporter();
+    public static final String PATH_TO_FILE = "src/test/resources/net/sf/jabref/exporter/";
+    public String xmlFileName;
+
+    @Parameter
+    public String filename;
+
+    @Rule
+    public TemporaryFolder testFolder = new TemporaryFolder();
+
+
+    @Parameters(name = "{0}")
+    public static Collection<String> fileNames() throws IOException {
+        List<String> files = new ArrayList<>();
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(Paths.get(PATH_TO_FILE))) {
```

Comment:
I think this is kind of double code:
Your first collect all files and then filter on them, but you are only searching for some specific files:
You could instead use: Stream stream = Files.list(Paths.get) and then apply the filters directly on it or you directly use Files.find(...) to only find the bib files

Have a look here at Listing Files / Finding Files:
http://winterbe.com/posts/2015/03/25/java8-examples-string-number-math-files/